File: server/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/chat/ChatRange.java
package calebxzhou.rdi.mc.common2.chat;

import java.util.Locale;

public enum ChatRange {
    HOST("本房间"),
    GLOBAL("公共");

    private final String displayName;

    ChatRange(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static ChatRange fromRcmdValue(String value) {
        return switch (value.toLowerCase(Locale.ROOT)) {
            case "global" -> GLOBAL;
            default -> HOST;
        };
    }
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/chat/ChatRange.kt
package calebxzhou.rdi.mc.rcmd.chat

enum class ChatRange(val displayName: String) {
    HOST("本房间"),
    GLOBAL("公共");


    companion object {
        fun fromRcmdValue(value: String): ChatRange {
            return when (value.lowercase()) {
                "global" -> ChatRange.GLOBAL
                else -> ChatRange.HOST
            }
        }
    }
}
